bypass_url_parser.py

```diff
@@ -262,13 +262,22 @@ def _generate_curls(self, url_obj: ParseResult):
 
         # [http_headers_scheme] - Custom scheme rewrite with X-Forwarded-Scheme
         if any(mode in ["all", "http_headers_scheme"] for mode in self.current_bypass_modes):
-            for const_header_scheme in self.const_header_schemes:
-                for const_proto in self.const_protos:
-                    cmd = [*self.base_curl, "-H", f"{const_header_scheme}: {const_proto}", target_url]
-                    item = CurlItem(url_obj, self.base_curl, cmd, bypass_mode="http_headers_scheme",
-                                    target_ip=url_public_ip, debug=self.debug, ext_logger=self.logger)
-                    if item not in self.curl_items:
-                        self.curl_items.append(item)
+            commands = set()
+            for const_proto in self.const_protos:
```

Comment:
Feels awful, but I'm not sure there's a better way for us to achieve this -.-

Reply:
Hehe, I know! The `set` of `tuple` is however handy here to reduce the code size:
- `set.add()` ensures duplicate eviction without having to do each time: `if element not in list, list.append...`;
- On small collection `set` is not significantly slower than list while iterating.

However, `set` requires immutable elements (commands are list and `set(list())` is impossible) and that's why I use `tuple` instead here ^_^

Small idea to make this a little less dirty (in next PR): move `commands` in `self.commands` in `init()` method and make a `self.command.clear()` here.

Reply:
My bad, this comment was meant for the line
```python
commands.add(tuple([*self.base_curl, "-H", f"Forwarded: proto={const_proto}", target_url]))
```
but for the "proto=" and "on ..." lines, http discremancies.. :shrug:

Reply:
OK, while waiting to find a cleaner way, no modification.